src/main/scala/nl/knaw/dans/easy/fedora2vault/FoXml.scala

```diff
@@ -66,24 +65,22 @@ object FoXml {
 
   def managedStreamLabel(foXml: Node, id: String): Option[String] = {
     getStreamRoot(id, foXml)
-      .withFilter(controlGroup(_).contains("M"))
+      .withFilter(hasControlGroup("M"))
       .flatMap(getLabel)
   }
 
-  private def controlGroup(streamRoot: Node): Option[String] = {
-    streamRoot.attribute("CONTROL_GROUP").map(_.text)
+  private def hasControlGroup(controlGroup: String)(streamRoot: Node): Boolean = {
+    streamRoot.attribute("CONTROL_GROUP").map(_.text).contains(controlGroup)
   }
 
   def getLabel(streamRoot: Node): Option[String] = {
-    (streamRoot \ "datastreamVersion")
-      .flatMap(_.attribute("LABEL"))
-      .flatten.headOption.map(_.text)
+    (streamRoot \ "datastreamVersion" \ "@LABEL").headOption.map(_.text)
```

Comment:
```suggestion
    streamRoot \ "datastreamVersion" \@ "LABEL"
```

Reply:
Thus you don't see a difference with a missing attribute or an empty attribute.
That's OK for `(prop \@ "NAME") endsWith "#ownerId"` but less so in this context.